calculator/main.py

```diff
@@ -58,18 +85,29 @@ def get_history(calc):
         ie:
         ('2016-05-20 12:00:00', 'add', (1, 2), 3),
     """
-    pass
+    return calc['history']
 
 
 def reset_history(calc):
     """
     Resets the calculator history back to an empty list.
     """
-    pass
+    calc['history'] = []
 
 
 def repeat_last_operation(calc):
     """
     Returns the result of the last operation executed in the history.
     """
-    pass
+    # Check if the history is empty
+    if calc['history']:
+        # Get the last history entry and bits from it we need
+        latest_index = len(calc['history'])-1
+        latest_history_entry = calc['history'][latest_index]
+        latest_operation = latest_history_entry[1]
+        latest_params = latest_history_entry[2]
+        
+        result = perform_operation(calc, latest_operation, latest_params)
+        return(result)
+    else:
+        return None
```

Comment:
calc['history'] is already a list of lists. You can reach the last operation by simply indexing the last item of the last list:

    return calc['history'][-1][-1]